src/sql/workbench/contrib/executionPlan/browser/widgets/findExpensiveNodeWidget.ts

```diff
@@ -0,0 +1,204 @@
+	private _operationNameSelectBox: SelectBox;
+	private _selectedExpensiveOperationType: ExpensiveOperationType = ExpensiveOperationType.Cost;
+
+	constructor(
+		public readonly widgetController: ExecutionPlanWidgetController,
+		public readonly executionPlanDiagram: AzdataGraphView,
+		@IContextViewService public readonly contextViewService: IContextViewService,
+		@IThemeService public readonly themeService: IThemeService,
+		@INotificationService public readonly notificationService: INotificationService
+	) {
+		super(DOM.$('.find-expensive-operation-widget'), 'findExpensiveOperation');
+		this.renderAndStyleWidget();
+	}
+
+	private renderAndStyleWidget() {
+		// Expensive Operation Dropdown
+		this._operationNameSelectBoxContainer = DOM.$('expensive-operation-name-select-box .dropdown-container');
+		this.container.appendChild(this._operationNameSelectBoxContainer);
+
+		this._operationNameSelectBox = new SelectBox([
+			ACTUAL_ELAPSED_TIME_STRING,
+			ACTUAL_ELAPSED_CPU_TIME_STRING,
+			COST_STRING,
+			SUBTREE_COST_STRING,
+			ACTUAL_NUMBER_OF_ROWS_FOR_ALL_EXECUTIONS_STRING,
+			NUMBER_OF_ROWS_READ_STRING
+		], COST_STRING, this.contextViewService, this._operationNameSelectBoxContainer);
+		this._operationNameSelectBox.render(this._operationNameSelectBoxContainer);
+		attachSelectBoxStyler(this._operationNameSelectBox, this.themeService);
+		this._operationNameSelectBoxContainer.style.width = '150px';
+
+		this._operationNameSelectBox.onDidSelect(e => {
+			switch (e.selected) {
+				case ACTUAL_ELAPSED_TIME_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.ActualElapsedTime;
+					break;
+				case ACTUAL_ELAPSED_CPU_TIME_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.ActualElapsedCpuTime;
+					break;
+				case COST_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.Cost;
+					break;
+				case SUBTREE_COST_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.SubtreeCost;
+					break;
+				case ACTUAL_NUMBER_OF_ROWS_FOR_ALL_EXECUTIONS_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.ActualNumberOfRowsForAllExecutions;
+					break;
+				case NUMBER_OF_ROWS_READ_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.NumberOfRowsRead;
+			}
+		});
+
+		// Apply Button
+		const self = this;
+		this._operationNameSelectBox.selectElem.onkeydown = async (ev) => {
+			if (ev.key === 'Enter') {
+				await new FindExpensiveOperationAction().run(self);
+			}
+			else if (ev.key === 'Escape') {
+				this.widgetController.removeWidget(self);
+			}
+		};
+
+		const findButton = new Button(this.container, {
+			title: localize('findExpensiveOperationButtonTitle', 'Find Expensive Operation (Enter)')
+		});
+		findButton.setWidth('60px');
+		findButton.label = localize('findExpensiveOperationApplyButton', 'Find');
+
+		findButton.onDidClick(async e => {
```

Comment:
The widget should be disposable and all disposable objects it creates should be registered. 

(I'd probably make the base widget itself extend Disposable - it looks like you aren't doing this for any of the other widgets either)